### ISSUE TITLE
Villager Trade Discusion

### ISSUE THREAD

**Issue by CovertJaguar** [Railcraft/Railcraft#851]
So I'm in the middle or rewriting the villager trades and I'm at a bit of a loss on how distribute the trades between villager levels.

There are basically four categories of trade items atm:
- Raw materials (coal, coke, rails, metals) - buy and sell
- Tracks - sell (too many types?)
- Minecarts/Locos - sell
- Tools (crowbar, overalls, goggles, tuners, etc...) - sell

It seems to me that generally, raw material trades are more useful, but that the other trades help introduce the players to new things or avoid annoying crafting chains.

It also feels like the trades should be leveled in terms of complexity and uniqueness:
1. Raw Materials
2. Tracks
3. Minecarts
4. Tools

But when doing this, if you were to do a lot of trades (with for example the Trade Station), that the more useful trades (raw mats) would be crowded out by the less flexible trades at higher levels.

Its also entirely possible that I have a basic misunderstanding of how trading works.

So I am at a loss as what to do with this. Input?

**Comment by Speiger** [comment]
Well @CovertJaguar i do not think that this would be to crouded. Since 1.7.10 the trade system got changed quite a bit. There are a lot more trades to get by default so that is not something you need to worry.
About the system why dont you make a villager profssion for each of the categories?
It would keep things sorted & you could say that these things are possible to get when you trade with them. (Some are 100% chance other less)
I hope that helps...

**Comment by maruohon** [comment]
So I'm assuming that by "level" you mean the vanilla tier system in 1.8+? Or is this issue still directed at 1.7.10? Well I'll give some of my views for 1.8+ and the new trading system anyway. I haven't actually looked much into the code side of things in 1.8+, so this is based in game-play + wiki information.

I don't know how the Trade Station works, but in the vanilla-like trading system, you could have one or several basic trades on the first tiers. When the player trades with the villager, he unlocks the subsequent tiers one by one. You should be able to determine how many trades you want each new tier to have, and they will be randomly selected when the tier unlocks/generates the trades.

So in my opinion, you should be able to have more of the lower-tier trades, and then just one or two (or however many you want per villager) of the higher tier ones. That way the vanilla-style trading menu doesn't get too crowded with things, and the lower-tier material trades (more useful?) are also first in the list. (Since in 1.8+ the tiers unlock in order [new ones go to the end of the list like before] and thus the entire trade list isn't completely random like it used to be in 1.7.10 and before).

Hopefully this didn't completely miss the point...

**Comment by bookerthegeek** [comment]
Multiple villagers would likely be simpler.

**Comment by Ristellise** [comment]
You could do that... oh if possible could you allow for custom ores to be bought at the rarity of the ore itself?

**Comment by Forecaster** [comment]
@Joshwoo70 What do you mean by "custom ores"? Do you mean Railcraft ores?

**Comment by CovertJaguar** [comment]
I'm not opposed to adding ore trading.

**Comment by Ristellise** [comment]
@Forecaster no .. include all the ores found in a game. Iron ore lead ore silicon ore etc etc.
TLDR add modded ores that are frohnd in the game install in ore dict

**Comment by CovertJaguar** [comment]
Uh no, I'd stick to Railcraft ores.

**Comment by Ristellise** [comment]
@CovertJaguar Fine with me too.

**Comment by liach** [comment]
Villagers should never trade ore... They should process steel for you.

**Comment by liach** [comment]
I did not add ores but rather added raw materials, tracks, and carts. The trade station itself needs a redesign, too, so that factor is not considered. See #1180.
